src/operator/join.rs

```diff
@@ -651,10 +704,8 @@ where
     It: IntoIterator<Item = (Z::Key, Z::Val)> + 'static,
 {
     fn eval(&mut self, index: &I, trace: &T) -> Z {
-        /*println!("JoinTrace::eval@{}:\n  index:\n{}\n  trace:\n{}",
-        self.time,
-        textwrap::indent(&index.to_string(), "    "),
-        textwrap::indent(&trace.to_string(), "    "));*/
+        self.stats.lhs_tuples += index.len();
+        self.stats.rhs_tuples += trace.len();
```

Comment:
This is probably not a good metric.  `index` is a batch of changes and it makes sense to sum them up, but `trace` stores accumulated changes of the rhs stream, so it doesn't make sense to sum them up again.

Reply:
What would you do instead?

Reply:
Perhaps just: `self.stats.rhs_tuples = trace.len();` (and we don't need to invoke it here for each input batch, just once in `fn metadata()`).

Reply:
We don't have access to the trace in `.metadata()` so I may just assign it every time, shouldn't really matter that much

Reply:
Right, I wasn't thinking.